### ISSUE TITLE
Poetry do not install package version with a post suffix, even in exact requirements

### ISSUE THREAD

**Issue by 7sDream** [python-poetry/poetry#3657]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: macOS 11.1
- **Poetry version**: 1.1.4
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: N/A, a empty project is enough to reproduce this issue

## Issue

Try use this command to install nonebot2 package, version `2.0.0a9.post1`(comply with PEP440), which is [exist](https://pypi.org/project/nonebot2/2.0.0a9.post1/) in pypi:

`poetry add nonebot2@2.0.0a9.post1 --allow-prereleases`

but get version `2.0.0.a9` installed:

![image](https://user-images.githubusercontent.com/7822577/107148196-f44da800-698c-11eb-8538-3cf50080f946.png)

And pyproject.toml says version is `2.0.0a9.post1`:

```toml
[tool.poetry]
name = "poetrytest"
version = "0.1.0"
description = ""
authors = ["7sDream <i@7sdre.am>"]

[tool.poetry.dependencies]
python = "^3.9"
nonebot2 = {version = "2.0.0a9.post1", allow-prereleases = true}

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

I can use pip to install/upgrade it to target version, though:

![image](https://user-images.githubusercontent.com/7822577/107148213-13e4d080-698d-11eb-8281-a9625f0aafa8.png)

![image](https://user-images.githubusercontent.com/7822577/107148219-16dfc100-698d-11eb-91e2-498840d0b81e.png)

Any ideas?

**Comment by finswimmer** [comment]
Hello @7sDream,

thanks for reporting. The problem here is not the `.post` release per-se. The problem is, that it is a post release of a prerelease, which poetry currently doesn't handle correct.

fin swimmer

**Comment by radoering** [comment]
Should be fixed via python-poetry/poetry-core#327.

**Comment by dimbleby** [comment]
confirmed that with latest master code from poetry and poetry-core indeed version 2.0.0a9.post1 is installed, this can be closed
